src/offset_of.rs

```diff
@@ -94,6 +95,30 @@ macro_rules! offset_of {
     }};
 }
 
+#[cfg(feature = "unstable_const")]
+#[macro_export(local_inner_macros)]
+macro_rules! offset_of {
+    ($parent:path, $field:tt) => {{
+        _memoffset__field_check!($parent, $field);
+
+        // Get a base pointer.
+        // No UB here, and the pointer does not dangle, either.
+        let uninit = $crate::mem::MaybeUninit::<$parent>::uninit();
+        unsafe {
+            // This, on the other hand, *is* UB, since we're creating a reference
+            // to uninitialized data.
+            // Unfortunately it's the best we can do at the moment; but at least we
+            // restrict this UB to happen at compile-time.
```

Comment:
Well that comment is not true any more now.

Reply:
Yeah, you're right. Doesn't that mean that the constant-enabled macro is less sound than the previous macro?

Reply:
Well, it is not much worse. The stable macro has one instance of "reference to invalid value", the const macro has two instances of that.